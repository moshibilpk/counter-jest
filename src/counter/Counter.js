import { View, Text, StyleSheet } from 'react-native'
import React from 'react'

export default function Counter({ count, incrementCount, decrementCount }) {

    return (
        <View style={styles.container}>
            <Text testID='increment' style={styles.counterButton} onPress={incrementCount}>
                Increment
            </Text>
            <Text style={styles.counter}>{count}</Text>
            <Text testID='decrement' style={styles.counterButton} onPress={decrementCount}>
                Decrement
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    counterButton: {
        backgroundColor: '#000',
        color: '#fff',
        padding: 10,
        borderRadius: 5
    },
    counter: {
        fontSize: 20,
        color: '#000',
        margin: 5
    }
})