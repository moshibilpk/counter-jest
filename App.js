import { View, Text } from 'react-native'
import React, { useState } from 'react'
import Counter from './src/counter/Counter'

export default function App() {
  const [count, setCount] = useState(0)

  const incrementCount = () => {
    setCount(previousCount => previousCount + 1)
  }

  const decrementCount = () => {
    count === 0 ? 0 : setCount(previousCount => previousCount - 1)
  }

  return (
    <>
      <Counter
        count={count}
        setCount={setCount}
        incrementCount={incrementCount}
        decrementCount={decrementCount}
      />
    </>
  )
}