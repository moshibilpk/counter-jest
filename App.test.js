// @flow
import React from 'react'
// import renderer from 'react-test-renderer'
import { Text } from 'react-native';

import Counter from './src/counter/Counter'
import App from './App'
import { findAllWithType, findWithType, isComponentOfType } from 'react-shallow-testutils';
import ShallowRenderer from 'react-shallow-renderer';
// import ShallowRenderer from 'react-test-renderer/shallow';


const renderApp = () => {
    renderer = new ShallowRenderer();
    renderer.render(
        <App />
    );
    return renderer.getRenderOutput();
};

// console.log("ðŸš€ ~ file: Counter.test.js ~ line 21 ~ renderApp ~ renderApp", result)
describe('counter test correctly', () => {
    const result = renderApp()
    console.log("ðŸš€ ~ file: Counter.test.js ~ line 26 ~ describe ~ result", result)

    test('0 visible in count', () => {
        const result = renderApp()
        expect(result).toMatchSnapshot()
        expect(findWithType(result, Counter).props.count).toEqual(0);
        // console.log("ðŸš€ ~ file: Counter.test.js ~ line 28 ~ test ~ result", result)
        // const test = findAllWithType(wrapper, Counter);
        // expect(findAllWithType(wrapper, Counter))
        console.log("ðŸš€ ~ file: Counter.test.js ~ line 22 ~ test ~ findAllWithType(wrapper, Counter)", findAllWithType(result, Counter, 'increment'))
        // const data = wrapper('0')
    })

    // test('Increment button visible', () => {
    //     const result = renderApp()
    //     expect(result).toMatchSnapshot()
    //     expect(findWithType(result, Counter).props.count).toEqual(0);

    // });
    
})