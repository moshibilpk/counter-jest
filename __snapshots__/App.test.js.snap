// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter test correctly 0 visible in count 1`] = `
<React.Fragment>
  <Counter
    count={0}
    decrementCount={[Function]}
    incrementCount={[Function]}
    setCount={[Function]}
  />
</React.Fragment>
`;
